packages/protocol/test/signal/SignalService.integration.test.ts

```diff
@@ -0,0 +1,242 @@
+        const {
+            l1Provider,
+            owner,
+            l1SignalService,
+            l2SignalService,
+            srcChainId,
+            headerSync,
+        } = await deployIntegrationSignalService();
+
+        const signal = ethers.utils.hexlify(ethers.utils.randomBytes(32));
+
+        const tx = await l1SignalService.connect(owner).sendSignal(signal);
+        await tx.wait();
+
+        const app = ethers.constants.AddressZero;
+        const key = await l1SignalService.getSignalSlot(app, signal);
+
+        const { block, blockHeader } = await getBlockHeader(l1Provider);
+        await headerSync.setSyncedHeader(block.hash);
+
+        const signalProof = await getSignalProof(
+            l1Provider,
+            l1SignalService.address,
+            key,
+            block.number,
+            blockHeader
+        );
+
+        await expect(
+            l2SignalService.isSignalReceived(
+                srcChainId,
+                app,
+                signal,
+                signalProof
+            )
+        ).to.be.revertedWith("B:app");
+    });
+
+    it("should revert since signal == HashZero", async function () {
+        const {
+            l1Provider,
+            owner,
+            l1SignalService,
+            l2SignalService,
+            srcChainId,
+            headerSync,
+        } = await deployIntegrationSignalService();
+
+        const signal = ethers.utils.hexlify(ethers.utils.randomBytes(32));
+
+        const tx = await l1SignalService.connect(owner).sendSignal(signal);
+        await tx.wait();
+
+        const app = owner.address;
+        const key = await l1SignalService.getSignalSlot(app, signal);
+
+        const { block, blockHeader } = await getBlockHeader(l1Provider);
+        await headerSync.setSyncedHeader(block.hash);
+
+        const signalProof = await getSignalProof(
+            l1Provider,
+            l1SignalService.address,
+            key,
+            block.number,
+            blockHeader
+        );
+
+        await expect(
+            l2SignalService.isSignalReceived(
+                srcChainId,
+                app,
+                ethers.constants.HashZero,
+                signalProof
+            )
+        ).to.be.revertedWith("B:signal");
+    });
+
+    it("should pass and return true", async function () {
+        const {
+            l1Provider,
+            owner,
+            l1SignalService,
+            l2SignalService,
+            srcChainId,
+            headerSync,
+        } = await deployIntegrationSignalService();
+
+        const signal = ethers.utils.hexlify(ethers.utils.randomBytes(32));
+
+        const tx = await l1SignalService.connect(owner).sendSignal(signal);
+        await tx.wait();
+
+        const app = owner.address;
+        const key = await l1SignalService.getSignalSlot(app, signal);
+
+        const { block, blockHeader } = await getBlockHeader(l1Provider);
+        await headerSync.setSyncedHeader(block.hash);
```

Comment:
Please try to call `isSignalReceived` before `setSyncedHeader`, then call again after, to verify if synced header is invalid, then the first `isSignalReceived` will fail.